fork-common/src/main/java/com/shazam/fork/suite/TestSuiteLoader.java

```diff
@@ -0,0 +1,111 @@
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+public class TestSuiteLoader {
+    private static final String TEST_ANNOTATION = "Lorg/junit/Test;";
+    private static final String IGNORE_ANNOTATION = "Lorg/junit/Ignore;";
+
+    private final File instrumentationApkFile;
+    private final DexFileExtractor dexFileExtractor;
+    private final TestClassMatcher testClassMatcher;
+
+    public TestSuiteLoader(File instrumentationApkFile, DexFileExtractor dexFileExtractor, TestClassMatcher testClassMatcher) {
+        this.instrumentationApkFile = instrumentationApkFile;
+        this.dexFileExtractor = dexFileExtractor;
+        this.testClassMatcher = testClassMatcher;
+    }
+
+    public Collection<TestCaseEvent> loadTestSuite() throws NoTestCasesFoundException {
+        List<TestCaseEvent> testCaseEvents = dexFileExtractor.getDexFiles(instrumentationApkFile).stream()
+                .map(dexFile -> dexFile.ClassDefsSection.getItems())
+                .flatMap(Collection::stream)
+                .filter(c -> testClassMatcher.matchesPatterns(c.getClassType().getTypeDescriptor()))
+                .map(this::convertClassToTestCaseEvents)
+                .flatMap(Collection::stream)
+                .collect(toList());
+
+        if (testCaseEvents.isEmpty()) {
+            throw new NoTestCasesFoundException("No tests cases were found in the test APK: " + instrumentationApkFile.getAbsolutePath());
+        }
+        return testCaseEvents;
+    }
+
+    @Nonnull
+    private List<TestCaseEvent> convertClassToTestCaseEvents(ClassDefItem classDefItem) {
+        AnnotationDirectoryItem annotationDirectoryItem = classDefItem.getAnnotations();
+        if (annotationDirectoryItem == null) {
+            return emptyList();
+        }
+
+        List<TestCaseEvent> testCaseEvents = new ArrayList<>();
+        for (AnnotationDirectoryItem.MethodAnnotation method : annotationDirectoryItem.getMethodAnnotations()) {
+            asList(method.annotationSet.getAnnotations()).stream()
+                    .filter(annotation -> TEST_ANNOTATION.equals(stringType(annotation)))
+                    .map(annotation -> convertToTestCaseEvent(classDefItem, annotationDirectoryItem, method, annotation))
+                    .forEach(testCaseEvents::add);
+        }
+        return testCaseEvents;
+    }
+
+    @Nonnull
+    private TestCaseEvent convertToTestCaseEvent(ClassDefItem classDefItem,
+                                                 AnnotationDirectoryItem annotationDirectoryItem,
+                                                 AnnotationDirectoryItem.MethodAnnotation method,
+                                                 AnnotationItem annotation) {
+        String testMethod = method.method.getMethodName().getStringValue();
+        String testClass = getClassName(classDefItem);
+        boolean ignored = isClassIgnored(annotationDirectoryItem) || isMethodIgnored(annotation);
```

Comment:
This ```ignored``` boolean might be wrong, as the previous filter in ```convertClassToTestCaseEvents``` removes anything but ```TEST_ANNOTATION```.   That said, no _ignore_ ```annotation```  can be present here. 

I think we can relax the filter in the method ```convertClassToTestCaseEvents```    to something like:
```
.filter(annotation -> (TEST_ANNOTATION.equals(stringType(annotation))
                            || (IGNORE_ANNOTATION.equals(stringType(annotation)))))
```
to fix this.